src/components/Kv/KvCarousel.vue

```diff
@@ -29,53 +166,288 @@ export default {
 		}
 
 		// setup autoplay
-		if (this.slides.length > 1 && this.autoplay) {
-			this.interval = setInterval(() => {
-				this.advance();
-			}, 5000);
+		if (this.autoplay && this.slides.length > 1) {
+			// init the CSS custom property for CSS animation
+			this.$refs.KvCarousel.style.setProperty(
+				'--kv-carousel-autoplay-interval', `${this.autoplayInterval}s`
+			);
+
+			const refreshRate = 0.1; // 100 milliseconds
+			this.intervalTimerCurrentTime = 0;
+			this.intervalTimer = setInterval(() => {
+				if (!this.paused) {
+					this.intervalTimerCurrentTime += refreshRate;
+				}
+				if (this.intervalTimerCurrentTime >= this.autoplayInterval) {
+					this.intervalTimerCurrentTime = 0;
+					this.advance();
+				}
+			}, refreshRate * 1000); // 100 milliseconds
 		}
 	},
 	beforeDestroy() {
-		clearInterval(this.interval);
+		if (this.autoplay) {
+			clearInterval(this.intervalTimer);
+		}
 	},
 	methods: {
+		/**
+		 * Move forward one slide
+		 *
+		 * @public This is a public method
```

Comment:
These @public decorations should be removed since these methods will not be called by reference anymore (they are only called by reference from KivanSlider component)

Reply:
My thinking was they could be useful in the future if other components on a page want to interact with this slider. Don't feel strongly though. Also I was trying to get Storybook to auto-document the methods but no luck :/